pkg/api/filters/headers/validator.go

```diff
@@ -0,0 +1,56 @@
+package headers
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/Azure/open-service-broker-azure/pkg/api/filters"
+	log "github.com/Sirupsen/logrus"
+)
+
+var responseMissingAPIVersion = []byte(
+	`{ "error": "MissingAPIVersion", "description": "The request did not ` +
+		`include the X-Broker-API-Version header"}`,
+)
+
+var responseAPIVersionIncorrect = []byte(
+	`{ "error": "APIVersionIncorrect", "description": "X-Broker-API-Verson ` +
+		`header includes an incompatible version"}`,
+)
+
+type validator struct{}
+
+//NewValidator creates a new instance of the header validator
+func NewValidator() filters.Filter {
+	return &validator{}
+}
+
+//Filter validates that required headers are present
+func (v *validator) Filter(handler http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		apiVersion := r.Header.Get("X-Broker-API-Version")
+		if apiVersion == "" {
+			v.sendError(w, responseMissingAPIVersion)
+			return
+		}
+		if !strings.HasPrefix(apiVersion, "2.") {
```

Comment:
This is the same thing that Pivotal does in their go broker api library. We can refine this to a specific version if we'd like.